src/problem3.rb

```diff
@@ -0,0 +1,23 @@
+class Problem3
+  def getDuplicatedValues(values)
+    response = []
+    values.each do |value|
+      if(value.abs() > values.count)
+        if values[value.abs()-1 - values.count] > 0
+          values[value.abs()-1 - values.count] = values[value.abs()-1 - values.count] * -1
+        elsif values[value.abs()-1 - values.count] > -values.count
+          response.push value.abs()
+          values[value.abs()-1 - values.count] = values[value.abs()-1 - values.count]
+        end
+      else
+        if values[value.abs()-1] > 0
+          values[value.abs()-1] = values[value.abs()-1] * -1
+        elsif values[value.abs()-1] > -values.count
+          response.push value.abs()
+          values[value.abs()-1] = values[value.abs()-1] - values.count
+        end
+      end
+    end
+    return response
```

Comment:
Still not right,

A couple of things:

- `Given an array of numbers having size N in which every number is between 1 and N, determine if there are any duplicates in it` so you are suppose to return a boolean not a array. The key to every good solution is understanding the problem.

- There a couple of foul code smells about your solution, 'Conditional Complexity',  'Oddball Solution' read more about it here https://blog.codinghorror.com/code-smells/

PS: you don't need a `return` in ruby.

Reply:
Really? Then it's easy, I was just refactoring on how to return the values, so that i will return unique values for the duplicates. Damn. Will clean it up.